Docs se16536 sqs dh (#862)
mulesoft/docs-connectors: amazon-sqs/5.6/modules/ROOT/pages/amazon-sqs-connector-config-topics.adoc

= Amazon SQS Connector 5.6 Additional Configuration Information - Mule 4
:page-aliases: connectors::amazon/amazon-sqs-connector-config-topics.adoc

> 

Reviewer: I would add like an intro sentence to reference the upcoming configuration sections, something like:
_Set additional configurations for Anypoint Connector for Amazon SQS (Amazon SQS Connector), such as how to receive messages in cluster mode and create the FIFO queue_....

[node-behavior]
== Receiving Messages in Cluster Mode

> The default behavior of the Receive Messages input source is to receive messages on all nodes when running in cluster mode. To have the app receive messages on the primary node only, select the *Primary node only* checkbox on the *Advanced* sub-tab on the *Receive messages* tab.

Reviewer: ```suggestion
The default behavior of the Receive Messages input source is to receive messages on all nodes when running in cluster mode. To have the app receive messages on the primary node only:
. In Studio, select the Receive Messages input source of your flow.
.  On the *Advanced* sub-tab of the *Receive messages* tab, select the *Primary node only* checkbox.
```

The multi-node cluster behavior depends on the selected SQS queue type. When using the Standard queue, SQS guarantees that a message is delivered at least once. However, duplicate messages can be introduced in the queue. You can use a FIFO queue to ensure that a message is delivered only once and that it remains available until a consumer processes and deletes it.

For more information about queue types, see https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/standard-queues.html[AWS SQS Stanard Queues] and https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/FIFO-queues.html[AWS SQS FIFO Queues].

== Creating the FIFO Queue

> 

Reviewer: ```suggestion
FIFO (First-In-First-Out) queues deliver messages in the order in which they are sent.
```

Author: Nice to add this

> To create the FIFO queue, use the *Create queue* operation in the connector, and set the *FifoQueue* and *ContentBasedDeduplication* field values to true.
> 
> End the *queue name* value with the `.fifo` extension: for example, `MyTestFIFOQueue.fifo`.
> 
> While creating the FIFO queues, choose either of the two regions that Amazon supports:

Reviewer: ```suggestion
To create the FIFO queue:
. In Studio, add the *Create queue* operation in your flow.
. In the operation properties, set the *FifoQueue* and *ContentBasedDeduplication* field values to `true`.
. End the *queue name* value with the `.fifo` extension,  for example `MyTestFIFOQueue.fifo`.
. Choose either of the two regions that Amazon supports: +
* US_EAST_2 (US East Ohio)
* US_WEST_2 (US West Oregon)
```

Reviewer: Not sure if these are the correct steps, since I can not test the connector in Studio to see it work. For example, Where is *queue name* field? and where can the user enter the Amazon regions?

Reviewer 2: Good question. I don't see fields that match FifoQueue and ContentBasedDeduplication. Those looks like XML values anyway. We should update to match the Studio UI rather than use XML values.

Author: This is a good find, but it will take some time to play with the functionality to check this section. I will capture your comments and will  log  a JIRA for investigating and rewording the Creating FIFO Queue section.
